src/nccl_ofi_interface_neuron.c

```diff
@@ -4,9 +4,21 @@
 
 #include "config.h"
 
+#include <stdlib.h>
 #include "nccl_ofi.h"
 #include "nccl_ofi_api.h"
 
+static ncclResult_t init_v4(ncclDebugLogger_t logFunction)
+{
+	/* v4 API may block infinitively when executed with RDMA
+         * protocol because communicator creation is (a) blocking
+         * operation by definition of v4 interface and (b) performing
+         * 4-way handshake in case of RDMA protocol. Therefore, we
+         * force it to use SENDRECV protocol. */
+	setenv("OFI_NCCL_PROTOCOL", "SENDRECV", 0);
```

Comment:
In case the user have explicitly set `OFI_NCCL_PROTOCOL` to "RDMA", we should allow the v4 connect()/accept()/listen() APIs to explicitly fail with their appropriate error message as introduced in commit `a725e08a7b55 ("api: fail when using connect/accept_v4 with RDMA protocol")`.

Therefore, I think you should set here protocol to be SENDRECV only in case the user haven't set it already.

Reply:
Applied

Reply:
I'm not sure why we did this.  `setenv(..., 0)` will not change the environment variable if they key already exists in the environment.  So the `getenv()` is redundant here.